src/posix/platform/daemon.cpp

```diff
@@ -28,7 +28,7 @@
 
 #include "posix/platform/daemon.hpp"
 
-#if defined(__ANDROID__) && !OPENTHREAD_CONFIG_ANDROID_NDK_ENABLE
```

Comment:
Are we still using `OPENTHREAD_CONFIG_ANDROID_NDK_ENABLE` anywhere?
Can we remove it from set of configs?

Reply:
It's still used in the implementation of version string.
https://github.com/openthread/openthread/blob/9d852a3b370c6b45df25a6d6c5e8e7ad5297fc89/src/core/api/instance_api.cpp#L154

And there seems to be a platform `android-ndk` where it's enabled:
https://github.com/openthread/openthread/blob/9d852a3b370c6b45df25a6d6c5e8e7ad5297fc89/script/cmake-build#L160

This PR itself is targeted to fix a misconfiguration so maybe we can consider unifying the options in a future refactor PR.